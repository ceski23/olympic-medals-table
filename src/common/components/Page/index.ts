export { Page } from "./Page";
export type { CountryMedalsData } from "./Page";
