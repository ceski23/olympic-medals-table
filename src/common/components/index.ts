export * from "./Select";
export * from "./Input";
export * from "./Page";
export * from "./App";
