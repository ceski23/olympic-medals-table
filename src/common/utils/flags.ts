export const flags = Object.fromEntries(
  Object.entries(
    import.meta.glob("../../../node_modules/country-flag-icons/3x2/*.svg", {
      as: "url",
      eager: true,
    })
  ).map(([id, url]) => [id.match(/([A-Z][A-Z])\.svg/)?.[1] ?? id, url])
);
