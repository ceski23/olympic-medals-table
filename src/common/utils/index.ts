export * from "./guards";
export * from "./styles";
export * from "./flags";
