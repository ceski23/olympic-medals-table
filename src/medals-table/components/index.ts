export * from "./CountryFlag";
export * from "./MedalsTable";
export * from "./SortingIndicator";
export * from "./Table";
export * from "./CountryForm";
