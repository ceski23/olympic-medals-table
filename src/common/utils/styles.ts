import { Theme } from "@emotion/react";
import { Breakpoint } from "common/styles";

export const mediaQueryUp =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (min-width: ${theme.breakpoints[breakpoint] + 0.01}px)`;

export const mediaQueryUpTo =
  (breakpoint: Breakpoint) =>
  ({ theme }: { theme: Theme }) =>
    `@media (max-width: ${theme.breakpoints[breakpoint]}px)`;
