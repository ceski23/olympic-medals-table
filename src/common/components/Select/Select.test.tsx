import { screen } from "@testing-library/react";

import { Select } from "common/components";
import { renderWithProviders } from "test/utils";

describe("Select", () => {
  it("should render correctly", async () => {
    const { asFragment } = renderWithProviders(
      <Select id="test" label="Test" error="Error">
        <option>1</option>
        <option>2</option>
      </Select>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should display error state", () => {
    renderWithProviders(
      <Select id="test" label="Test" error="Error">
        <option>1</option>
        <option>2</option>
      </Select>
    );

    const input = screen.getByRole("combobox", { name: "Test" });
    expect(input).toBeInvalid();

    const errorText = screen.getByRole("alert");
    expect(errorText).toHaveTextContent("Error");
  });
});
