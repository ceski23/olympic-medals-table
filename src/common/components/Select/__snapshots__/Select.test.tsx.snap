// Vitest Snapshot v1

exports[`Select > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100%;
}

.emotion-2 {
  font-size: 0.9rem;
  margin-bottom: 5px;
  color: #707070;
}

.emotion-4 {
  border-radius: 5px;
  border: 1px solid #b6b6b6;
  padding: 10px 15px;
  font-size: 1rem;
  box-shadow: 3px 3px 7px #dddddd;
  min-width: 0;
}

.emotion-4 option:disabled {
  color: #707070;
}

.emotion-6 {
  color: red;
  font-size: 0.8rem;
  margin-top: 5px;
}

<div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      for="test"
    >
      Test
    </label>
    <select
      aria-describedby="test-error"
      aria-invalid="true"
      class="emotion-4 emotion-5"
      id="test"
    >
      <option>
        1
      </option>
      <option>
        2
      </option>
    </select>
    <div
      class="emotion-6 emotion-7"
      id="test-error"
      role="alert"
    >
      Error
    </div>
  </div>
</DocumentFragment>
`;
